examples/stable_fluid.py

```diff
@@ -123,21 +125,29 @@ def advect_semilag(vf: ti.template(), qf: ti.template(),
 
 
 @ti.kernel
-def advect_bfecc(vf: ti.template(), qf: ti.template(), new_qf: ti.template()):
+def advect_bfecc(vf: ti.template(), qf: ti.template(), new_qf: ti.template(), intermedia_qf: ti.template()):
     ti.cache_read_only(qf, vf)
     for i, j in vf:
         p = ti.Vector([i, j]) + 0.5
-        p_mid = backtrace(vf, p, dt)
-        q_mid = bilerp(qf, p_mid)
-        p_fin = backtrace(vf, p_mid, -dt)
-        q_fin = bilerp(qf, p_fin)
-        new_qf[i, j] = q_mid + 0.5 * (q_fin - qf[i, j])
+        p = backtrace(vf, p, dt)
+        intermedia_qf[i, j] = bilerp(qf, p)
+
+    ti.cache_read_only(intermedia_qf, qf, vf)
+    for i, j in vf:
```

Comment:
Is the key problem about separating bfecc to two offload?

Reply:
Yes. To be more precise, we need a forward advection (backtrack + interpolate) to get an inter-media quantity field first, then a backward advection to (backtrace in -dt direction + interpolate) to get the field at the original time stamp (assuming there is no error).

Since for the second one (backward advection) we need the whole inter-media field ready, it's necessary to run it in a separate another loop and store it already.

Reply:
But to be noticed, `jacobbi_dual` caused serious artifacts with the new `bfecc`.

I will post detailed comparisons between combination of different methods when I get a minute.